### ISSUE TITLE
Unhandled Exception: saving changes resource reference editor

### ISSUE THREAD

**Issue by ghramsey** [MattFiler/OpenCAGE#185]
Trying to changen the vdu setting for a rewire panel per helpful tips in #181 , but when I tried to click the save changes button it throws a general exception.

This is the first map I'd changed other than original Tech_hub map.

![image](https://user-images.githubusercontent.com/51859157/233766387-27524b63-6d09-443f-bef1-cdaffc11e5ae.png)

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.Exception: Tried to initialise ShortGuid without 4-byte ID string.
   at CATHODE.Scripting.ShortGuid..ctor(String id)
   at CommandsEditor.Popups.UserControls.GUI_Resource_CollisionMapping.get_CollisionID()
   at CommandsEditor.AddOrEditResource.SaveChanges_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4614.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
CommandsEditor
    Assembly Version: 0.2.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///E:/SteamLibrary/steamapps/common/Alien%20Isolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4550.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4605.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4390.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
CathodeLib
    Assembly Version: 0.4.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///E:/SteamLibrary/steamapps/common/Alien%20Isolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
netstandard
    Assembly Version: 2.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/netstandard/v4.0_2.0.0.0__cc7b13ffcd2ddd51/netstandard.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4605.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
PresentationCore
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4605.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
----------------------------------------
WindowsBase
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4605.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------
Newtonsoft.Json
    Assembly Version: 13.0.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///E:/SteamLibrary/steamapps/common/Alien%20Isolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Numerics.Vectors
    Assembly Version: 4.1.4.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///E:/SteamLibrary/steamapps/common/Alien%20Isolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
websocket-sharp
    Assembly Version: 1.0.2.19803
    Win32 Version: 0.2.0.0
    CodeBase: file:///E:/SteamLibrary/steamapps/common/Alien%20Isolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box. 
```

**Comment by ghramsey** [comment]
To add to this.  I tried this change on a different file. Tech_RND.  One vdu setting saved the change the other one also threw 
an exception.  Is is possible that this change is not identical between maps?

**Comment by MattFiler** [comment]
I think this is probs an issue with the collision mapping info - try removing that and see if it lets you save?

Thanks for the report!

**Comment by ghramsey** [comment]
There's no collision mapping selected for either of these two vdu settings.  
And I don't know which option in thelist is appropriate to select.

VDU_20_25cm04 lets me save the change using the Save all Changes button; 
VDU_20_25cm03 throws the error for same change.

By comparison for the Tech_Hub map this list box has selected an option.
So the lack of selection seems to be, or is related to, the error?  Given the text of the error:

` 
************** Exception Text **************
System.Exception: Tried to initialise ShortGuid without 4-byte ID string.
   at CATHODE.Scripting.ShortGuid..ctor(String id)  `

In the end I'm not overly concerned with changing this setting.  

I happened to be blanket editing the maps to speed things up and 
this was one of the things I'd listed to change.  
I don't think I'll be doing too much with the rewire panels here for now.
